js/components/portal-dashboard/all-responses-popup/student-responses-popup.tsx

```diff
@@ -29,32 +29,34 @@ interface IProps {
 interface IState {
   selectedStudents: Map<any, any>[];
   showSpotlightDialog: boolean;
+  showSpotlightListDialog: boolean;
 }
 export class StudentResponsePopup extends React.PureComponent<IProps, IState> {
   constructor(props: IProps) {
     super(props);
     this.state = {
       selectedStudents: [],
-      showSpotlightDialog: false
+      showSpotlightDialog: false,
+      showSpotlightListDialog: false
     };
   }
   render() {
     const { anonymous, currentActivity, currentQuestion, hasTeacherEdition, isAnonymous, onClose, questions,
             setAnonymous, setCurrentActivity, setStudentFilter, sortedQuestionIds, studentCount, students,
             toggleCurrentQuestion, trackEvent } = this.props;
-    const { selectedStudents, showSpotlightDialog } = this.state;
+    const { selectedStudents, showSpotlightDialog, showSpotlightListDialog } = this.state;
     return (
       <div className={css.popup} data-cy="all-responses-popup-view">
         <PopupHeader currentActivity={currentActivity} onCloseSelect={onClose} />
         <div className={css.tableHeader}>
           <PopupClassNav
             anonymous={anonymous}
-            isSpotlightOn={showSpotlightDialog && selectedStudents.length > 0}
+            isSpotlightOn={selectedStudents.length > 0}
             studentCount={studentCount}
             setAnonymous={setAnonymous}
             setStudentFilter={setStudentFilter}
             trackEvent={trackEvent}
-            onShowDialog={this.showSpotlightDialog(true)}
+            onShowDialog={selectedStudents.length > 0 ? this.setShowSpotlightListDialog(true) : this.setShowSpotlightDialog(true)}
```

Comment:
Calling `setState()` in render is an anti-pattern. Ideally, you should set the state outside of any render-related function, but if necessary you can do so in `componentDidUpdate()` with a test so that you don't call `setState()` when the value hasn't changed. Otherwise, there's potential for infinite render loops.

Reply:
Hmm... I think this is a little deceptive.  With the function currying used on `setShowSpotlightListDialog` and `setShowSpotlightDialog` we're not actually setting state here, but instead using the ternary to selectively pass one of two callback functions into `PopupClassNav` that can be used when needed by the component (and the currying let's us specify an argument).  Then again, I might be missing something.  However, I'm not sure that the currying is needed anymore.  This work was massively refactored last night, so I'm not sure that we need this extra layer of complexity anymore.

Reply:
You're absolutely right -- I missed the currying completely. That said, it looks like you never pass the false version of the curried function, so it seems like you could eliminate the boolean argument and just pass the function.

Reply:
I pushed a commit that I think should make this clearer.  It removed the function currying on `setShowSpotlightListDialog` an `setShowSpotlightDialog` so that it is more clear what is happening with these functions when we pass them to `PopupClassNav` via props.  Again, I don't believe that either of these cases involved a state change in the render function, but perhaps I'm missing something.

Reply:
Also, @kswenson is it ok if I add you as reviewer for future Portal Dash PRs?  We're slim on reviewers on this project, and some of these PRs could use an extra set of eyes.

Reply:
Sure, I've generally been reading through them anyway. 😉

Reply:
The false version was being passed, but to a different component.  Regardless, I think it's a little easier to read in its present form.

Reply:
The `PopupClassNav` only has the job of showing the components, whereas the components themselves contain the x close button and have the job of closing themselves.